stackit/internal/services/iaas/securitygrouprule/resource.go

```diff
@@ -0,0 +1,697 @@
+				},
+			},
+			"icmp_parameters": schema.SingleNestedAttribute{
+				Description: "ICMP Parameters.",
+				Optional:    true,
+				Attributes: map[string]schema.Attribute{
+					"code": schema.Int64Attribute{
+						Description: "ICMP code. Can be set if the protocol is ICMP.",
+						Optional:    true,
+						PlanModifiers: []planmodifier.Int64{
+							int64planmodifier.RequiresReplace(),
+						},
+						Validators: []validator.Int64{
+							int64validator.AtLeast(0),
+							int64validator.AtMost(255),
+						},
+					},
+					"type": schema.Int64Attribute{
+						Description: "ICMP type. Can be set if the protocol is ICMP.",
+						Optional:    true,
+						PlanModifiers: []planmodifier.Int64{
+							int64planmodifier.RequiresReplace(),
+						},
+						Validators: []validator.Int64{
+							int64validator.AtLeast(0),
+							int64validator.AtMost(255),
+						},
+					},
+				},
+			},
+			"ip_range": schema.StringAttribute{
+				Description: "The remote IP range which the rule should match.",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					validate.IP(),
+				},
+			},
+			"port_range": schema.SingleNestedAttribute{
+				Description: "The range of ports.",
+				Optional:    true,
+				Computed:    true,
+				Attributes: map[string]schema.Attribute{
+					"max": schema.Int64Attribute{
+						Description: "The maximum port number. Should be greater or equal to the minimum.",
+						Optional:    true,
+						Computed:    true,
+						PlanModifiers: []planmodifier.Int64{
+							int64planmodifier.UseStateForUnknown(),
+							int64planmodifier.RequiresReplace(),
+						},
+						Validators: []validator.Int64{
+							int64validator.AtLeast(0),
+							int64validator.AtMost(65535),
+						},
+					},
+					"min": schema.Int64Attribute{
+						Description: "The minimum port number. Should be less or equal to the minimum.",
+						Optional:    true,
+						Computed:    true,
+						PlanModifiers: []planmodifier.Int64{
+							int64planmodifier.UseStateForUnknown(),
+							int64planmodifier.RequiresReplace(),
+						},
+						Validators: []validator.Int64{
+							int64validator.AtLeast(0),
+							int64validator.AtMost(65535),
+						},
+					},
+				},
+			},
+			"protocol": schema.SingleNestedAttribute{
+				Description: "The internet protocol which the rule should match.",
+				Optional:    true,
+				Computed:    true,
+				Attributes: map[string]schema.Attribute{
+					"name": schema.StringAttribute{
+						Description: "The protocol name which the rule should match.",
+						Optional:    true,
+						Computed:    true,
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.UseStateForUnknown(),
+							stringplanmodifier.RequiresReplace(),
+						},
+					},
+					"protocol": schema.Int64Attribute{
```

Comment:
would it make more sense to name this field `number` instead? `protocol.protocol` path seems weird

Reply:
rename to number, they will probably do the same in the API

Reply:
I renamed the sub `protocol` field to `number`. 
In api it is still `protocol`, that is why some places it still looks like `protocol.protocol`. But from terraform implementation side, it is `protocol.number`